ingestion-beam/src/main/java/com/mozilla/telemetry/republisher/RepublishPerDocType.java

```diff
@@ -22,9 +22,11 @@ public static RepublishPerDocType of(RepublisherOptions baseOptions) {
 
   @Override
   public PDone expand(PCollection<PubsubMessage> input) {
-    List<Destination> destinations = baseOptions.getPerDocTypeEnabledList().stream()//
-        .map(Destination::new) //
+    List<Destination> destinations = baseOptions.getPerDocTypeDestinations().entrySet().stream()
+        .flatMap(
+            entry -> entry.getValue().stream().map(value -> new Destination(entry.getKey(), value)))
```

Comment:
This looks correct and in line with the existing implementation. Reading through this class again, though, I realize we should probably be doing something more efficient.

Currently, we create a separate pcollection for each doctype appearing in the values; that would mean potentially we would be creating multiple sinks to the same Pub/Sub topic for multiple different docTypes. Ideally, we should be creating one pcollection per output topic. I'm also realizing we should probably support allowing a single docType to appear in multiple output topics, which would mean not using `Partition`.

These are out of scope here. If you'd be willing to write these up in a follow-up issue, though, I'd appreciate it. This work does not seem worth prioritizing until we have use cases where it actually matters.